plugins/i3s/io/EsriReader.cpp

```diff
@@ -0,0 +1,435 @@
+*     * Neither the name of Hobu, Inc. or Flaxen Geo Consulting nor the
+*       names of its contributors may be used to endorse or promote
+*       products derived from this software without specific prior
+*       written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+* FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+* COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+* INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+* BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+* OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+* AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+* OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+* OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+* OF SUCH DAMAGE.
+****************************************************************************/
+
+#include "EsriReader.hpp"
+#include "../lepcc/src/include/lepcc_c_api.h"
+#include "../lepcc/src/include/lepcc_types.h"
+#include "pool.hpp"
+#include "SlpkExtractor.hpp"
+
+#include <istream>
+#include <cstdint>
+#include <cstring>
+#include <cmath>
+#include <cstdio>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <chrono>
+#include <pdal/util/FileUtils.hpp>
+#include <pdal/util/ProgramArgs.hpp>
+#include <pdal/util/Bounds.hpp>
+#include <pdal/pdal_features.hpp>
+#include <pdal/compression/LazPerfCompression.hpp>
+#include <gdal.h>
+
+namespace pdal
+{
+    void EsriReader::addArgs(ProgramArgs& args)
+    {
+        args.add("bounds", "Bounds of the point cloud", m_args.bounds);
+        args.add("threads", "Number of threads to be used."
+                "This number will be squared", m_args.threads);
+    }
+
+    void EsriReader::initialize(PointTableRef table)
+    {
+        Json::Value config;
+        if (log()->getLevel() > LogLevel::Debug4)
+            config["arbiter"]["verbose"] = true;
+        m_arbiter.reset(new arbiter::Arbiter(config));
+
+        const std::string pre("i3s://");
+        if (Utils::startsWith(m_filename, pre))
+            m_filename = m_filename.substr(pre.size());
+
+        if (m_filename.back() == '/')
+            m_filename.pop_back();
+
+        log()->get(LogLevel::Debug) << "Fetching info from " << m_filename <<
+            std::endl;
+
+        try
+        {
+            initInfo();
+        }
+        catch (std::exception& e)
+        {
+            throw pdal_error(std::string("Failed to fetch info: ") + e.what());
+        }
+
+        //create pdal Bounds
+        m_bounds = createBounds();
+        //find number of nodes per nodepage
+        m_nodeCap = m_info["store"]["index"]["nodesPerPage"].asInt();
+
+        log()->get(LogLevel::Debug) << "\n\nFileName: "
```

Comment:
Styling consistency - put the `<<` at the end of the previous line instead of at the start of lines.  Also remove the `\n` characters from these logged strings - we're already using `std::endl` so skip the extra whitespace.